.buildkite/build-and-test.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash -u
+
```

Comment:
Lifted the logic below from WordPress iOS and updated the paths and step names for how this repo is configured.